compiler/src/iree/compiler/Codegen/Common/OptimizeTensorInsertExtractSlices.cpp

```diff
@@ -116,6 +116,10 @@ hoistLoopInvariantSubsetAtIterArg(RewriterBase &rewriter,
       return loopLike;
     loopLike = *newLoop;
 
+    // Replace hoisted out extract with corresponding induction variable.
+    rewriter.replaceAllUsesExcept(
+        extraction.getResult(), loopLike.getRegionIterArgs().back(), loopLike);
```

Comment:
Will the replaced iter_arg always be the last iter arg? If so, can you write docs on why?

Reply:
Yeah since the newLoop generated always add the replaced/hoisted extraction as the new IV. But I just found out they had a flag/bool as part of that API anyways. 

Now just modified the flag to do the right thing lol!

Reply:
FYI: https://github.com/iree-org/iree/pull/17713#discussion_r1651713642 @MaheshRavishankar

Reply:
Read the comments on that previous PR. Based on my reading of the upstream implementation, this flag really just does what my previous fix does which is to replace all the user of extraction with the approriate induction variable(IV).

I do see some reordering of IV after this pass, but it still maintain/match the correct yield, iter_arg, and users(both inside and outside payload). Additionally, this phenomenon is unrelated to the flag but rather caused by the way we loop through the region iter_args from the caller fn and doing the replaceWithAdditionalYield. 

Plus I think we should be kind of safe because we have already used this flag before in [here](https://github.com/iree-org/iree/blob/907b2cdbafb3f69666695b4e66b737932b6eb71b/compiler/src/iree/compiler/Codegen/Common/HoistUnrolledVectorExtractInsertSlice.cpp#L145-L147) :)